babel/numbers.py

```diff
@@ -1112,7 +1117,7 @@ def apply(
 
         retval = ''.join([
             self.prefix[is_negative],
-            number,
+            number if self.number_pattern != '' else '',
```

Comment:
If the distinction between `None` and `''` isn't significant, maybe just
```suggestion
            (number if self.number_pattern else ''),
```

Reply:
Ok, just to be sure, this assumes we are not concerned about people manually constructing `NumberPattern` on their own, since that could make it a breaking change.

Eg. if someone does `NumberPattern(pattern, (pos_prefix, neg_prefix),
                         (pos_suffix, neg_suffix), grouping,
                         int_prec, frac_prec,
                         exp_prec, exp_plus)` (note, no number pattern), then `number_pattern` will be `None` and this `NumberPattern` won't show numbers anymore even though it used to.

Differentiating here is the reason I used None rather than empty string as the default.